src/som/vmobjects/SInteger.java

```diff
@@ -24,19 +25,38 @@
 
 package som.vmobjects;
 
-import java.math.BigInteger;
-
 import som.vm.Universe;
 
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+
 public final class SInteger extends SNumber {
 
+  /**
+   * Language convention requires integers up to this value to be identical.
+   */
+  private static final long MAX_IDENTICAL_INT = 1073741823L;
```

Comment:
Actually, IntegerTest#equalityAndIdentityTest was broken, which led me to implementing this change. The requirements are immediately visible from the test source. These are where I picked the value from.

As for the benchmarks, I have applied these changes with the idea in mind that the SOM baseline is still supposed to be simple rather than super fast, to allow for experimentation with optimisations. Has the philosophy changed during the past years?

I am going to move forward with merging the PR, this discussion notwithstanding.